samples/client/petstore/scala-sttp/src/main/scala/org/openapitools/client/model/User.scala

```diff
@@ -27,6 +27,6 @@ case class User(
   phone: Option[String] = None,
   /* User Status */
   userStatus: Option[Int] = None
-) extends ApiModel
+) extends ApiModel with ExceptionApiModel
```

Comment:
and howcome user is part of `ExceptionApiModel`? Like when you create new user and post is unsuccessful you will return existing user in response body?

Reply:
I didn't find any example on how to enrich the model metadata with the info on whether or not it is used for processing 4xx codes. That's why all models are also `ExceptionApiModel`.

Reply:
@michalo2882 could you please provide an example how to map response into results?
Let me give an example with models i expect:
```scala
case class Pet(...) extends ApiModel
case class PetBadRequest(code: Int, message: String) extends ExceptionApiModel
case class PetValidationFailed(code: Int, message: String, errors: Seq[String]) extends ExceptionApiModel
```
and spec mapping:
```
200 -> Pet
400 -> PetNotFound
422 -> PetValidationFailed
```

I expect something like this
```scala
val api = new PetApi(...)
val pet = Pet(...)

val response = api.addPet(pet)

response.mapResponse { 
  case Right(createdPet) => createdPet
  case Left(responseException: ResponseException[PetBadRequest, Exception]) => (response.model.code, response.model.message)
  case Left(responseException: ResponseException[PetValidationFailed, Exception]) => response.model.errors.foreach(println)
  case Left(responseException: DeserializationException[Exception]) => responseException.message
  case Left(responseException: HttpError[Exception]) => responseException.statusCode
}
```